internal/osutil/mount_linux.go

```diff
@@ -30,3 +68,60 @@ func IsMounted(baseDir string) (bool, error) {
 	}
 	return false, nil
 }
+
+// Mount attaches a filesystem accessible via the device node specified by the source
+// to the specified target. If not existing, target will be created before
+// mounting the filesystem.
+func Mount(opts *MountOptions) error {
+	if err := os.MkdirAll(opts.Target, 0755); err != nil {
+		return fmt.Errorf("cannot create directory %q: %w", opts.Target, err)
+	}
+	flags := uintptr(0)
+	if opts.ReadOnly {
+		flags |= unix.MS_RDONLY
+	}
+	if opts.Remount {
+		flags |= unix.MS_REMOUNT
+	}
+	if err := syscallMount(opts.Source, opts.Target, opts.MountType, flags, ""); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Unmount removes the attachment of the topmost filesystem mounted on the specified target.
+func Unmount(opts *UnmountOptions) error {
+	if opts.Force {
+		// Force unmount without taking care of flushing data to the disk
+		if err := syscallUnmount(opts.Target, unix.MNT_FORCE); err != nil {
+			return err
+		}
+		return nil
+	}
+	entries, err := LoadMountInfo(procSelfMountInfo)
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		if opts.Target == entry.MountDir {
+			syscallSync()
+			// Attempt to remount as read-only
```

Comment:
I don't know much about mounting/unmounting, so just curious: why do we need to remount as read-only before unmounting?

Reply:
Because sometimes you could call `umount(2)` on a filesystem that's currently in use and disrupt other programs' presumably critical work. In fact, as it is currently implemented right now, unmounting root file system before shutdown in Pebble itself would result in a system with the rootfs unmounted but still live (since `/sbin/shutdown` can't possibly be invoked)